apps/oxlint/src/lint.rs

```diff
@@ -326,6 +358,26 @@ impl LintRunner {
             Err(error)
         }
     }
+
+    fn adjust_ignore_patterns(base: &str, path: &str, ignore_patterns: Vec<String>) -> Vec<String> {
+        if base == path {
+            ignore_patterns
+        } else {
+            let relative_ignore_path =
+                pathdiff::diff_paths(path, base).unwrap_or_else(|| PathBuf::from("."));
```

Comment:
side note: we can remove the dependency when we expect that the config file is always inside the cwd.
(`path.strip_prefix(base)`)
But then this would not work correctly: `oxlint -c ../.oxlintrc.json` :/